py-polars/tests/unit/lazyframe/test_optimizations.py

```diff
@@ -0,0 +1,82 @@
+import polars as pl
+from polars.testing import assert_frame_equal
+
+
+def test_is_null_followed_by_all() -> None:
+    lf = pl.LazyFrame({"group": [0, 0, 0, 1], "val": [6, 0, None, None]})
+
+    expected_df = pl.DataFrame({"group": [0, 1], "val": [False, True]})
+    result_df = lf.group_by("group").agg(pl.col("val").is_null().all()).collect()
+
+    assert_frame_equal(expected_df, result_df, check_row_order=False)
+
```

Comment:
We should also add tests for 0-length series, especially for the ones involving `any/all`